### ISSUE TITLE
SpaceBeforeBrace rule not noticing a particular case

### ISSUE THREAD

**Issue by eqyiel** [brigade/scss-lint#845]
My colleague noticed a case that the `SpaceBeforeBrace` rule doesn't complain about:

```
$responsive: true !default;

@mixin bp($point) {
  @if $responsive == true{
    @if $point == XS {
      @media (min-width: 0) { @content; }
    }
    @if $point == S {
      @media (min-width: 414px) { @content; }
    }	@else if $point == M {
      @media (min-width: 40em) { @content; }
    }	@else if $point == L {
      @media (min-width: 64em) { @content; }
    }	@else if $point == XL {
      @media (min-width: 1024px) { @content; }
    }	@else if $point == XXL {
      @media (min-width: 1400px) { @content; }
    }
  } @else {
    @if $point == XS {
      @content;
    }
    @if $point == S {
      @content;
    }	@else if $point == M {
      @content;
    }	@else if $point == L {
      @content;
    }	@else if $point == XL {
      @content;
    }
  }
}
```

If you save this file to `fail.scss` and run `scss-lint fail.scss` no errors are printed:
```
% scss-lint -v
scss-lint 0.49.0
% scss-lint fail.scss && echo 'Squeaky clean!'
Squeaky clean!
```

According to the default setting of [`SpaceBeforeBrace`](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#spacebeforebrace), `@if $responsive == true{` should throw a warning.

Is this a known issue or limitation?

**Comment by eqyiel** [comment]
@sds awesome!
